### PR TITLE
JUnit assertThrows

### PR DESCRIPTION
**Pull request by nhojpatrick**
Code could be cleaner if it could be Java 1.8.

I could set `<maven.compiler.testSource>1.8</maven.compiler.testSource>`... thoughts???

### CODE REVIEW

**Comment by garydgregory** [comment]
I am ok with Java 8 but others might not be. May you please bring this up on the dev ML? It would be better if using Java 8 would bring benefits to the main code.

**Comment by nhojpatrick** [comment]
@garydgregory have continued assertThrows email about maven.compiler.testSource.

About the jdk 1.7 build failure, i'm not sure exactly what the issue is as from what i'm aware i've not touched anything to do with `japicmp/maven/JApiCmpMojo`

File: src/test/java/org/apache/commons/validator/routines/DomainValidatorStartupTest.java [hunk 1]
```diff
@@ -35,24 +39,52 @@
...
     public void testUpdateBaseArrayCC() {
-        DomainValidator.updateTLDOverride(ArrayType.COUNTRY_CODE_RO, new String[]{"com"});
+        // FIXME Simplification once upgraded to Java 1.8
+        final ThrowingRunnable testMethod = new ThrowingRunnable() {
+            public void run() {
+                DomainValidator.updateTLDOverride(ArrayType.COUNTRY_CODE_RO, new String[]{"com"});
+            }
+        };
+        final IllegalArgumentException thrown = assertThrows(IllegalArgumentException.class, testMethod);
+        assertThat(thrown.getMessage(), is(equalTo("Cannot update the table: COUNTRY_CODE_RO")));
```
Comment (garydgregory) [inline src/test/java/org/apache/commons/validator/routines/DomainValidatorStartupTest.java:51]: Same comment as in the Apache Commons Collections PR. Let's skip asserting the exception messages.

Comment (nhojpatrick) [reply src/test/java/org/apache/commons/validator/routines/DomainValidatorStartupTest.java:51]: Why skip? It's a common conversation I always keep having... I just wondered your rational...

My argument to alway checking the exception message is how do you know the test is failing as you are expecting it to fail?

You write a test to check X, and it throws Y exception. If Z might also causes Y to be thrown, so how do you know your testing X? My view is if you skip checking the exception message then you might as well just delete the test.

So Digester BeanPropertySetterRuleTestCase.testSetUnknownProperty according to the error messages and comments should produce an NoSuchMethodException, but currently is throws org.xml.sax.SAXException due to a code path meaning any non expected Exception is ignored.

So this test is to check COUNTRY_CODE_RO, but if the code behind the test changes and IllegalArgumentException is still thrown, how do you know it's the correct exception for what you have setup?

**Comment by nhojpatrick** [comment]
Could this get another review and try and get it merged.
I'm leaving asserting the exception messages, as it's good practice. You write a test with a set input, the output is also important.

File: pom.xml [hunk 2]
```diff
@@ -132,6 +132,18 @@
     </plugins>
   </build>
 
+  <dependencyManagement>
```
Comment (garydgregory) [inline pom.xml:135]: This is already defined in the parent POM. Is it not working for you?

Comment (nhojpatrick) [reply pom.xml]: i was rebasing 2-3 year old branches and had not spotted it move into the commons-parent. will correct all the branches i've made.

**Comment by nhojpatrick** [comment]
@garydgregory breaking it out into separate pr's, as i keep needing to rebase and often face merge issues, so easier if 1 pr per test. then it might be easier to get them reviewed and merged quicker.

https://github.com/apache/commons-validator/pull/92
https://github.com/apache/commons-validator/pull/93
https://github.com/apache/commons-validator/pull/94
https://github.com/apache/commons-validator/pull/95

**Pull request closed without merging**